packages/graphback-datasync/src/DataSyncPlugin.ts

```diff
@@ -151,38 +154,47 @@ export class DataSyncPlugin extends GraphbackPlugin {
     // Create Delta Type
     const modelName = model.graphqlType.name;
     const modelTC = schemaComposer.getOTC(modelName);
-    const allowedFieldEntries = Object.entries(modelTC.getFields()).filter((e: [string, ObjectTypeComposerFieldConfig<any, unknown, any>]) => {
-      // Remove relationship fields from delta Types
-      return parseRelationshipAnnotation(e[1].description) === undefined;
-    });
+    const TimestampSTC = schemaComposer.getSTC(GraphbackTimestamp.name);
 
-    const allowedFields = Object.assign({}, ...Array.from(allowedFieldEntries, ([k, v]: [string, any]) => ({ [k]: v })));
+    const DeltaTypeFields = this.getDeltaTypeFields(modelTC);
 
     // Add Delta Type to schema
-    schemaComposer.createObjectTC(getDeltaType(modelName)).addFields({
-      ...allowedFields,
-      [DataSyncFieldNames.deleted]: 'Boolean',
+    const DeltaOTC = schemaComposer.createObjectTC(getDeltaType(modelName)).addFields({
+      ...DeltaTypeFields,
+      [DataSyncFieldNames.deleted]: GraphQLBoolean,
     });
 
+
     // Create and Add Delta List type to schema
-    schemaComposer.createObjectTC({
+    const DeltaListOTC = schemaComposer.createObjectTC({
       name: getDeltaListType(modelName),
       fields: {
-        items: `[${getDeltaType(modelName)}]!`,
-        lastSync: GraphQLNonNull(GraphbackTimestamp)
+        items: GraphQLNonNull(GraphQLList(DeltaOTC.getType())),
+        lastSync: GraphQLNonNull(TimestampSTC.getType())
       }
     });
 
     // Add Delta Queries
     const deltaQuery = getDeltaQuery(model.graphqlType.name);
     schemaComposer.Query.addFields({
-      [deltaQuery]: `${getDeltaListType(modelName)}!`
+      [deltaQuery]: GraphQLNonNull(DeltaListOTC.getType())
     });
 
-    const findFilterTypeName = getInputTypeName(modelName, GraphbackOperationType.FIND);
+    const findFilterITC = schemaComposer.getITC(getInputTypeName(modelName, GraphbackOperationType.FIND));
     schemaComposer.Query.addFieldArgs(deltaQuery, {
-      lastSync: GraphQLNonNull(GraphbackTimestamp),
-      filter: findFilterTypeName
+      lastSync: GraphQLNonNull(TimestampSTC.getType()),
+      filter: findFilterITC.getType()
+    });
+  }
+
+  private getDeltaTypeFields(modelTC: ObjectTypeComposer) {
+    const DeltaTypeFieldEntries = Object.entries(modelTC.getFields()).filter((e: [string, ObjectTypeComposerFieldConfig<any, unknown, any>]) => {
```

Comment:
`ObjectTypeComposerFieldConfig` could probably be replaced with the GraphQL equivalent.

```suggestion
    const DeltaTypeFieldEntries = Object.entries(modelTC.getFields()).filter((e: [string, ObjectTypeComposerFieldConfig<any, unknown, any>]) => {
```